internal/app/referenceclient/client.go

```diff
@@ -216,9 +216,11 @@ func invoke(ctx context.Context, req *v1.ClientCompatRequest, trace *tracer.Trac
 	case v1.HTTPVersion_HTTP_VERSION_UNSPECIFIED:
 		return nil, errors.New("an HTTP version must be specified")
 	}
-	if trace != nil {
-		transport = tracer.TracingRoundTripper(transport, trace)
-	}
+
+	// Create a new TracingRoundTripper with our WireTracer so that the tests can trace values on the
+	// wire. Note that 'trace' could be nil, in which case, any error traces will
+	// simply not be printed.
+	transport = tracer.TracingRoundTripper(transport, NewWireTracer(trace))
```

Comment:
Note that this now creates the `TracingRoundTripper` every time since it's needed for the wire details. If `tracer` is nil, it just bypasses the trace printing.

Reply:
No better way; this is exactly what I would expect to see.